### ISSUE TITLE
Scrolling sticks to top when `"cursor_blink": false` set

### ISSUE THREAD

**Issue by baldwindavid** [zed-industries/zed#13720]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

This is the strangest thing, but just in the last weeks I've noticed that scrolling to the very top of a pane and then attempting to scroll down (via touchpad) results in a delay before scrolling down takes place. It takes multiple times before it reacts and the delay is up to a second. I originally assumed it was something related to my vim settings, but removed all settings except this and still get the behavior...

```json
{
    "cursor_blink": false
}
```

If I set that to `true`, regardless of settings, the problem disappears.

### Environment

Zed: v0.142.3 (Zed Preview)
OS: macOS 14.4.1
Memory: 32 GiB
Architecture: aarch64

**Comment by notpeter** [comment]
I can reproduce the scroll sticking independent of the `cursor_blink` setting. I just over-scroll with trackpad gestures (keep scrolling after the top is reached) and then try to scroll the other direction.

Interestingly this does not occur when I first open a project (e.g. while rust-analyzer loads the zed workspace).  Only once rust-analyzer is done indexing does the scroll lag behavior begin.

**Comment by baldwindavid** [comment]
I wonder if #11244 that @alygin and @SomeoneToIgnore were discussing is related.

**Comment by baldwindavid** [comment]
> I can reproduce the scroll sticking independent of the cursor_blink setting.

@notpeter Yes, I think I can too. It's just that it is a tiny delay for me as compared to an unusable near one second delay with `cursor_blink: false`.

**Comment by WeetHet** [comment]
Okay, so I'm not going insane and scrolling is indeed broken. That is nice to know, I guess. For me, scrolling breaks not only when over-scrolling, but also when I change focused panes or try to scroll a non focused pane. `cursor_blink: false` does not seem to fix it a lot, though it make the situation a bit better

**Comment by WeetHet** [comment]
@notpeter, good news, I've found the breaking PR (if I did bisect correctly):
https://github.com/zed-industries/zed/pull/13368

**Comment by WeetHet** [comment]
I guess we need @as-cii or @rtfeldman help here, because I have no idea what is going on in that pull request

**Comment by WeetHet** [comment]
Continuing on with my investigation, a bug is somewhere in the one of the two files: `crates/editor/src/element.rs` or `crates/editor/src/scroll/autoscroll.rs` because restoring them to `c58a8f1a04ba7ff56ec20ac0ad6fe909e3d3cf72` fixed the problem

**Comment by WeetHet** [comment]
I hate myself for doing this instead of just leaving this issue alone, but I went I found the exact lines causing the issue:
https://github.com/zed-industries/zed/blob/9b7bc04a87f3100c43e9ad036aaa73b73d06d5be/crates/editor/src/element.rs#L3660-L3663

Before #13368 they would execute even if `scroll_position` was equal to `current_scroll_position`. I don't know why they are equal in this case, so this might indicate a larger issue then I initially thought

**Comment by WeetHet** [comment]
@notpeter could you please reopen this issue, as it is not fixed for me yet (I've compiled zed from your branch and still experience scrolling sticking to the top)

**Comment by ChristoferMendes** [comment]
> @notpeter could you please reopen this issue, as it is not fixed for me yet (I've compiled zed from your branch and still experience scrolling sticking to the top)

I'm with the same problem here on Linux (with the build of the project available on curl), i.e.,

```
curl https://zed.dev/install.sh | sh  
```

**Comment by ChristoferMendes** [comment]
> > > @notpeter could you please reopen this issue, as it is not fixed for me yet (I've compiled zed from your branch and still experience scrolling sticking to the top)
> > 
> > I'm with the same problem here on Linux (with the build of the project available on curl), i.e.,
> > 
> > ```
> > curl https://zed.dev/install.sh | sh  
> > ```
> 
> Are you on Preview or Stable?

Stable

**Comment by notpeter** [comment]
> Stable. The scroll doesn't work at all. Not even clicking the scroll bar

@ChristoferMendes Can you open a new issue, I believe you are experiencing something distinct from this issue.  This is about a delay where scrolling lags temporarily after scrolling to the top of a file. Sounds like your scrolling is completely broken.

Separately, the fix for this issue went out in Zed Preview today (thanks @WeetHet for your help with #13874) and will be in Zed Stable next week (2024-07-17).
